### PR TITLE
feat: Introduce MaximumFollowNextPageLink option for usage with `Get-AzLogicAppRunHistory`

### PR DESCRIPTION
**Pull request by pim-simons**
Introduce MaximumFollowNextPageLink option in the config file for the Disable-AzLogicAppsFromConfig script.
This prevents retrieving all Logic App runs which can take a very long time. 

If the value is not supplied in the config file a default of 10 is used, 30 results are returned per call, so the default will return a max of 300 Logic App runs, which should be enough for most use-cases I think.

As mentioned here https://github.com/arcus-azure/arcus.scripting/pull/395

### CODE REVIEW

File: docs/preview/03-Features/powershell/azure-logic-apps.md [hunk 1]
```diff
@@ -177,6 +177,7 @@ Taking an example in which a set of Azure Logic Apps (`"rcv-shopping-order-*"`)
     "description": "Sender(s)",
     "checkType": "None",
     "stopType": "Immediate",
+	"maximumFollowNextPageLink": 25,
```
Comment (stijnmoreels) [inline docs/preview/03-Features/powershell/azure-logic-apps.md:180]:
```suggestion
    "stopType": "Immediate",
    "maximumFollowNextPageLink": 25,
```

File: docs/preview/03-Features/powershell/azure-logic-apps.md [hunk 2]
```diff
@@ -293,25 +294,23 @@ Typically done as the last task of the release pipeline, right after the deploym
...
 | ----------- | --------------- | --------------------------------------------------------------------------------------------------------------------- |
-| Description | `string`        | Description of Azure Logic App set to enable.                                                                        |
-| CheckType   | `enum`          | _Not taken into account for enabling Logic Apps._ |
-| StopType    | `enum`          | `None`: don't enable to given Logic Apps. |
-|             |                 | `Immediate`: enable the given Logic Apps. |
-| LogicApps   | `string array`  | Set of Logic App names to enable.                                                                                    |
-
+| Description | `string`        | Description of Azure Logic App set to enable.                                                                         |
+| CheckType   | `enum`          | _Not taken into account for enabling Logic Apps._                                                                     |
+| StopType    | `enum`          | `None`: don't enable to given Logic Apps.                                                                             |
```
Comment (stijnmoreels) [inline docs/preview/03-Features/powershell/azure-logic-apps.md:311]: This fix makes sure that it does not take forever to disable Logic Apps, can the same be said for enabling Logic Apps from a configuration file?

File: src/Arcus.Scripting.Tests.Unit/Arcus.Scripting.LogicApps.tests.ps1 [hunk 3]
```diff
@@ -280,6 +280,62 @@ InModuleScope Arcus.Scripting.LogicApps {
                 # Act
                 Disable-AzLogicAppsFromConfig -DeployFileName "$PSScriptRoot\Files\LogicApps\deploy-orderControl-noWaitingOrRunningRunsWithSingleImmediate.json" -ResourceGroupName $resourceGroup
 
+                # Assert
+                Assert-MockCalled Get-AzCachedAccessToken -Scope It -Times 1
+                Assert-MockCalled Disable-AzLogicApp -Scope It -Exactly 1 -ParameterFilter { $ResourceGroupName -eq $resourceGroup -and $LogicAppName -eq "snd-async" }
+            }
+            It "Disables all logic apps with option maximumFollowNextPageLink set to 25" {
```
Comment (stijnmoreels) [inline src/Arcus.Scripting.Tests.Unit/Arcus.Scripting.LogicApps.tests.ps1:287]: Good to have these tests as alternative to integration tests, as I'm guessing it is troublesome to set this up?

**Review by stijnmoreels — APPROVED** [review]
Again great stuff, descriptive tests with clear inputs. Super. 🎉

**Comment by stijnmoreels** [comment]
We can indeed, like you said, create a fix-release for this.
Thx for handling this issue, Pim!

File: docs/preview/03-Features/powershell/azure-logic-apps.md (hunk 2 above)
Comment (pim-simons) [reply docs/preview/03-Features/powershell/azure-logic-apps.md:311]: This is only applicable to where `Get-AzLogicAppRunHistory` is used. I found 2 other scripts so will make a change for that also 👍🏻

Comment (stijnmoreels) [reply docs/preview/03-Features/powershell/azure-logic-apps.md:311]: Super, thx! 👍

File: src/Arcus.Scripting.LogicApps/Scripts/Cancel-AzLogicAppRuns.ps1 [hunk 4]
```diff
@@ -1,10 +1,15 @@
...
     [Parameter(Mandatory = $true)][string] $ResourceGroupName = $(throw "Name of the resource group is required"),
-    [Parameter(Mandatory = $true)][string] $LogicAppName = $(throw "Name of the logic app is required")
+    [Parameter(Mandatory = $true)][string] $LogicAppName = $(throw "Name of the logic app is required"),
+    [Parameter(Mandatory = $false)][int] $MaximumFollowNextPageLink
 )
 
 try {
-    $runs = Get-AzLogicAppRunHistory -ResourceGroupName $ResourceGroupName -Name $LogicAppName -FollowNextPageLink | 
+    if (!$maximumFollowNextPageLink) {
+        $maximumFollowNextPageLink = 10
```
Comment (stijnmoreels) [inline src/Arcus.Scripting.LogicApps/Scripts/Cancel-AzLogicAppRuns.ps1:9]: Is this default also something we can place directly in the `param` block, no?

Comment (pim-simons) [reply src/Arcus.Scripting.LogicApps/Scripts/Cancel-AzLogicAppRuns.ps1]: You are right, see [`397cc66` (#396)](https://github.com/arcus-azure/arcus.scripting/pull/396/commits/397cc665970f3f3b79cf1dc2b440c07511bbad7e)

**Review by stijnmoreels — APPROVED** [review]
Great!

**Pull request merged**